src/libraries/System.Runtime.Numerics/src/System/Numerics/BigInteger.cs

```diff
@@ -2785,5 +2837,1166 @@ private void AssertValid()
                 Debug.Assert(_sign > int.MinValue);
             }
         }
+
+        //
+        // IAdditionOperators
+        //
+
+        /// <inheritdoc cref="IAdditionOperators{TSelf, TOther, TResult}.op_Addition(TSelf, TOther)" />
+        static BigInteger IAdditionOperators<BigInteger, BigInteger, BigInteger>.operator checked +(BigInteger left, BigInteger right) => left + right;
+
+        //
+        // IAdditiveIdentity
+        //
+
+        /// <inheritdoc cref="IAdditiveIdentity{TSelf, TResult}.AdditiveIdentity" />
+        static BigInteger IAdditiveIdentity<BigInteger, BigInteger>.AdditiveIdentity => Zero;
+
+        //
+        // IBinaryInteger
+        //
+
+        /// <inheritdoc cref="IBinaryInteger{TSelf}.DivRem(TSelf, TSelf)" />
+        public static (BigInteger Quotient, BigInteger Remainder) DivRem(BigInteger left, BigInteger right)
+        {
+            BigInteger quotient = DivRem(left, right, out BigInteger remainder);
+            return (quotient, remainder);
+        }
+
+        /// <inheritdoc cref="IBinaryInteger{TSelf}.LeadingZeroCount(TSelf)" />
+        public static BigInteger LeadingZeroCount(BigInteger value)
+        {
+            value.AssertValid();
+
+            if (value._bits is null)
+            {
+                return int.LeadingZeroCount(value._sign);
+            }
+
+            // When the value is positive, we just need to get the lzcnt of the most significant bits
+            // Otherwise, we're negative and the most significant bit is always set.
+
+            return (value._sign >= 0) ? uint.LeadingZeroCount(value._bits[^1]) : 0;
+        }
+
+        /// <inheritdoc cref="IBinaryInteger{TSelf}.PopCount(TSelf)" />
+        public static BigInteger PopCount(BigInteger value)
+        {
+            value.AssertValid();
+
+            if (value._bits is null)
+            {
+                return int.PopCount(value._sign);
+            }
+
+            ulong result = 0;
+
+            if (value._sign >= 0)
+            {
+                // When the value is positive, we simply need to do a popcount for all bits
+
+                for (int i = 0; i < value._bits.Length; i++)
+                {
+                    uint part = value._bits[i];
+                    result += uint.PopCount(part);
+                }
+            }
+            else
+            {
+                // When the value is negative, we need to popcount the two's complement representation
+                // We'll do this "inline" to avoid needing to unnecessarily allocate.
+
+                int i = 0;
+                uint part;
+
+                do
+                {
+                    // Simply process bits, adding the carry while the previous value is zero
+
+                    part = ~value._bits[i] + 1;
```

Comment:
Shouldn't this just be a _neg_? Seeing "~x + 1" in code feels odd to me.

Reply:
The underlying bits are `uint` and so negating converts it to a `long`, which is less efficient.

Reply:
According to my disasm the JIT doesn't optimize the pattern "~x + 1". If this is in a perf hot path (and it looks like _DangerousMakeTwosComplement_ might be?), then we should consider writing `var x = (uint)(-(int)value);`. Otherwise we should get the JIT to optimize "~x + 1".

Reply:
`~x + 1` is a basic optimization and the JIT should be taught to support it.

Reply:
Filed https://github.com/dotnet/runtime/issues/69003.